### ISSUE TITLE
Create package manager

### ISSUE THREAD

**Issue by hugomardbrink** [modmark-org/modmark#19]
**Is your feature request related to a problem? Please describe.**
When modules are available there needs to be a way for the packages to be stored and fetched in a standardised way. This should be done while compiling in the cli.

**Describe the solution you'd like**

- [x] #20
  - [ ] #51
  - [x] Package is local
  - [x] Package is in cache

- [x] Download packages in cache #48 

- [ ] Create an interface for the core to get modules from cli #47

**Comment by adelhult** [comment]
I've been thinking a bit more about how we can get the module manager to talk with the rest of the system. I suggest that we create a `Resolver` trait that looks something like this:

```rs
trait Resolve {
   fn resolve(path: &str) -> Option<&[u8]>;
}

// and then do
ModuleManager impl Resolve {
   ....
}
```

Then the `core::Context` object that handles all of the loaded modules could ask to get something that implements the resolve trait when creating it. Like so,
```rs
let mut ctx = Context::new(ModuleManager::new());
let output = eval(&source, &mut ctx);
``` 

A benefit of having a trait instead of always providing the `cli::ModuleManager` is that we for instance could handle module resolution in another simpler way if running inside of a web playground (or a safe mode that always denies external modules),

**Side note**:
I know that we have talked about whether or not the module imports should be handled by the parsing phase, the runtime or some other phase of the compilation. But I think this approach should work regardless and should be pretty easy to adapt regardless how we choose to handle it later. So I think we can deal with that issue later together in the whole group.

**Comment by CMDJojo** [comment]
Closed as of the merge of #169
